### PR TITLE
Fix texlive on darwin

### PR DESCRIPTION
**Pull request by mikesperber**
###### Motivation for this change

texlive-2016 would not build on Darwin.
###### Things done
- Unbreak build of clisp on Darwin (on which xindy depends on which texlive depends), which previously was Linux only.
- Add a dependency to xindy to libiconv.
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [x] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
ping @vbgl @matthewbauer @LnL7

**Comment by copumpkin** [comment]
cc @acowley

File: pkgs/development/interpreters/clisp/default.nix [hunk 1]
```diff
@@ -5,18 +5,18 @@
...
 , x11Support ? (stdenv.isi686 || stdenv.isx86_64)
 , dllSupport ? true
 , withModules ? [
-    "bindings/glibc"
     "pcre"
     "rawsock"
-    "wildcard"
-    "zlib"
   ]
+  ++ stdenv.lib.optional stdenv.isLinux [ "bindings/glibc" "zlib" "wildcard" ]
```
Comment (vcunat) [inline pkgs/development/interpreters/clisp/default.nix]: `optionals`, please, to avoid nested lists.

**Comment by mikesperber** [comment]
Thanks for the review! Done.

**Review by acowley — APPROVED** [review]

**Comment by vcunat** [comment]
It's not perfect, apparently: https://github.com/NixOS/nixpkgs/issues/20062

**Pull request merged**